modules/sdk-core/src/bitgo/utils/mpcUtils.ts

```diff
@@ -86,4 +87,37 @@ export abstract class MpcUtils {
     enterprise?: string;
     originalPasscodeEncryptionCode?: string;
   }): Promise<KeychainsTriplet>;
+
+  /**
+   * This function would be responsible for populating intents
+   * based on the type of coin / sig scheme the coin uses
+   * @param {IBaseCoin} baseCoin
+   * @param {PrebuildTransactionWithIntentOptions} params
+   * @returns {Record<string, unknown>}
+   */
+  populateIntent(baseCoin: IBaseCoin, params: PrebuildTransactionWithIntentOptions): Record<string, unknown> {
+    const chain = this.baseCoin.getChain();
+    const intentRecipients = params.recipients.map((recipient) => ({
+      address: { address: recipient.address },
+      amount: { value: `${recipient.amount}`, symbol: recipient.tokenName ? recipient.tokenName : chain },
+    }));
+
+    const baseIntent = {
+      intentType: params.intentType,
+      sequenceId: params.sequenceId,
+      comment: params.comment,
+      nonce: params.nonce,
+      recipients: intentRecipients,
+      memo: params.memo?.value,
+      token: params.tokenName,
+    };
+    if (baseCoin.getFamily() === 'eth') {
+      // Fill in intents for eth as it finalizes
+      return {
+        ...baseIntent,
```

Comment:
Updated, referred types from this link. But I think it makes sense to make the types from SDK as a single source of truth. Otherwise we would have to do copy pasting each time we update an intent

Reply:
agreed yeah having the types live in sdk makes more sense, we should open up a ticket for this